src/functions/welcome/welcome.go

```diff
@@ -0,0 +1,66 @@
+package welcome
+
+import (
+	"encoding/json"
+	"fmt"
+	"html"
+	"net/http"
+	"strings"
+	"os"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+// Data struct from Pubsub
+type PubSubMessage struct {
+	Data []byte `json:"data"`
+}
+
+type Payload struct {
+	UserId  string `json:"user_id"`
+	Username string `json:"username"`
+	ServerId string `json:"server_id"`
+	ServerName string `json:"server_name"`
+	AvatarUrl string `json:"avatar_url"`
+	ChannelId string `json:"channel_id"`
+}
+
+func GreetingNewMember(w http.ResponseWriter, r *http.Request) {
```

Comment:
We want a pubsub function and not an http, check `UpdateUserRole` function for ex

```go
func GreetingNewMember(ctx context.Context, m PubSubMessage) error {
...
    var payload Payload
	json.Unmarshal(m.Data, &payload)
}
```

Reply:
Done, I have pushed the change for it